src/formatFiles.js

```diff
@@ -1,27 +1,61 @@
 import { readFileSync, writeFileSync } from 'fs';
-import { resolveConfig, format } from 'prettier';
+import { resolveConfig, format as prettierFormat } from 'prettier';
 import { join } from 'path';
 
 export default (
-  directory,
+  rootDirectory,
   files,
   { config, onWriteFile, onExamineFile } = {}
 ) => {
+  // Use local "prettier-tslint" if possible.
+  const tslintFormat = loadFormat('prettier-tslint', 'format');
+  // Use local "prettier-eslint" if possible.
+  const eslintFormat = loadFormat('prettier-eslint');
+
   for (const relative of files) {
     onExamineFile && onExamineFile(relative);
-    const file = join(directory, relative);
-    const options = resolveConfig.sync(file, { config, editorconfig: true });
+    const file = join(rootDirectory, relative);
     const input = readFileSync(file, 'utf8');
-    const output = format(
-      input,
-      Object.assign({}, options, {
-        filepath: file,
-      })
-    );
+
+    const format =
+      (/\.(mjs|jsx?)$/.test(file) && (eslintFormat || tslintFormat)) ||
```

Comment:
This won't work for files with shebangs (which prettier supports). Maybe using `prettier.getFileInfo` would be better?

Reply:
Are you talking about executable files with no file extension that contain Javascript?

Reply:
Absolutely.

Reply:
https://github.com/prettier/prettier/pull/5149

Reply:
Let's support shebangs in a later patch. No biggie if they don't get linted for now.